Server/App/api/deck.py

```diff
@@ -256,10 +351,30 @@ def post(self, rating, user_id, connector, uuid):
         logging.info('{} rated deck {}'.format(user_id, uuid))
         return self.make_response(status=StatusCode.OK)
 
+class DeckShareHandler(BaseHandler):
+    @authorize_request_and_create_db_connector
+    @extract_user_id
+    def get(self, user_id, connector, uuid):
+        results = connector.call_procedure('FETCH_DECK', user_id, uuid)
+        if len(results) != 1:
+            return self.make_response(status=StatusCode.INTERNAL_SERVER_ERROR)
+        if results[0]['owner'] != user_id:
+            return self.make_response(status=StatusCode.FORBIDDEN)
+
+        while True:
```

Comment:
Ok that's true, I missed that. However, I think there will still be an issue because when there is a collision, it will return a single result of `null`, so `len(set_code) == 1`, even if a new code was not set.